ci.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/bash
+
+function display_usage {
+  echo "Usage: ci.sh [release|experimental[/m/model]]"
+  exit 1
+}
+
+#
+# Prevents 'clear' on exit of mingw64 bash shell
+#
+SHLVL=0
+
+#
+# Define paths; note Windows hosted bash assumptions for now
+#
+MODELROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+JQ="$MODELROOT/tools/jq-win64.exe"
+RSYNC="$MODELROOT/tools/rsync.exe"
+CI_CACHE="$MODELROOT/.cache"
+
+if [ ! -z "$SEVENZ_HOME" ]; then
+  APP7Z="$SEVENZ_HOME/7z"
+else
+  APP7Z="/c/Program Files/7-Zip/7z.exe"
+fi
+
+. "$MODELROOT/servervars.sh"
+. "$MODELROOT/resources/util.sh"
+. "$MODELROOT/resources/rsync-tools.sh"
+
+parse_args "$@"
+
+function run {
+
+  if [ ! -d "$CI_CACHE" ]; then
+    mkdir "$CI_CACHE"
+  fi
+
+  if [[ $DO_UPLOAD_ONLY == true ]]; then
+    if [ ! -d "$CI_CACHE/upload" ]; then
+      mkdir "$CI_CACHE/upload"
+    fi
+    rsync_to_downloads_keyman_com "$CI_CACHE/upload/" models/ true
+    exit 0
+  fi
+
+  if [[ $DO_ZIP_ONLY == true ]]; then
+    if [ ! -d "$CI_CACHE/data" ]; then
+      mkdir "$CI_CACHE/data"
+    fi
+    if [ ! -d "$CI_CACHE/upload" ]; then
+      mkdir "$CI_CACHE/upload"
+    fi
+    zip_model_info
+    rsync_to_downloads_keyman_com "$CI_CACHE/data/" model-data/
```

Comment:
Why is this rsync inside the `$DO_ZIP_ONLY` block?

Reply:
`$DO_ZIP_ONLY` might be a misnomer -- it means upload only the model metadata.

Reply:
This is also only really used for testing